### PR TITLE
[OSPP] Proposal for Edge Intelligence Benchmark for Edge-Cloud Collaborative Lifelong Detection

### PR DESCRIPTION
**Pull request by iszhyang**
Proposal for Edge Intelligence Benchmark for Edge-Cloud Collaborative Lifelong Detection

As a very promising community in edge AI Benchmarking, ianvs still has the following problems to be solved

The automatic dataset download and management features are missing, and users still need tedious operations to use the dataset.
Repeated deployment of the edge-cloud collaboration environment is too heavy. The distributed collaborative system simulation feature of ianvs needs to be developed urgently.
This proposal provides features in datasets, edge-cloud collaborative AI system simulation, etc. for the current version of Ianvs

### CODE REVIEW

File: docs/proposals/dataset_and_simulation.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  metrics:
+      # metric name; string type;
+    - name: "f1_score"
+      # the url address of python file
+      url: "./examples/pcb-aoi/singletask_learning_bench/testenv/f1_score.py"
+```
+
+For the convenience of developers, the expected testenv.yaml file is as follows:
+
+- The user fills in the `dataset_url`, here is the HUAWEI CLOUD OBS link.
```
Comment (Poorunga) [inline docs/proposals/dataset_and_simulation.md:59]: Does dataset url only support HUAWEI CLOUD obs service?

File: docs/proposals/dataset_and_simulation.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+![img](https://github.com/iszhyang/images/blob/main/ianvs/simulation.jpg?raw=true)
+
+- The `System Config` denotes the system config of the current test case, such as the `number of edge nodes`.
+- The `Simulation Controller` is the core module of system simulation, including `Paradigm Management` and `Enviroment Management`.
+  - `Paradigm Management` is used to manage the paradigm in simulation, including the following
+    - `Incremental Learning Management`
+    - `Lifelong Learning Management`
+    - etc.
+  - `Enviroment Management` is used to build and manage the simulation enviroments.
+    - `Enviroment Builder Controller` is used to parse the system config(simulation), build the simulation enviroment, and finally close the enviroment completely.
```
Comment (Poorunga) [inline docs/proposals/dataset_and_simulation.md:112]: Why is `System Config` not part of `Enviroment Builder Controller`?

File: docs/proposals/dataset_and_simulation.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  - generate the index file for train and test
+- dataset download link
+  - provide the download link of cityscapes dataset in HUAWEI Cloud OBS.
+- Baseline algorithm
+  ![](https://github.com/iszhyang/images/blob/main/ianvs/rfnet.png?raw=true)
+
+### Industrial Distributed Collaborative System Simulation
+
+Provide ianvs with the feature of industrial distributed system simulation.
+![img](https://github.com/iszhyang/images/blob/main/ianvs/simulation.jpg?raw=true)
```
Comment (MooreZheng) [inline docs/proposals/dataset_and_simulation.md:103]: As discussed, the architecture is overall fine but major revision is needed on the test case and container management of  simulation system. It is not easy work though.

File: docs/proposals/dataset_and_simulation.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  - `Paradigm Management` is used to manage the paradigm in simulation, including the following
+    - `Incremental Learning Management`
+    - `Lifelong Learning Management`
+    - etc.
+  - `Enviroment Management` is used to build and manage the simulation enviroments.
+    - `Enviroment Builder Controller` is used to parse the system config(simulation), build the simulation enviroment, and finally close the enviroment completely.
+    - `Container Management` manage the containers used to build the simulation enviroment.
+    - `Simulation Job Controller`is responsible for building and managing the job required in the simulation environment.
+
+![img](https://github.com/iszhyang/images/blob/main/ianvs/simulation-dataflow.jpg?raw=true)
```
Comment (MooreZheng) [inline docs/proposals/dataset_and_simulation.md:116]: Similarly, the dataflow is overall fine but major revision is needed on the test case and container management of  simulation system.

File: docs/proposals/dataset_and_simulation.md (hunk 1 above)
Comment (iszhyang) [reply docs/proposals/dataset_and_simulation.md]: I think the common OBS links all can be used to download.

File: docs/proposals/dataset_and_simulation.md (hunk 2 above)
Comment (iszhyang) [reply docs/proposals/dataset_and_simulation.md]: The `system config` here refers to the configuration file that the user needs to prepare in advance. So here I put it in place with `testing.yaml` and `algorithm.yaml`

**Comment by MooreZheng** [comment]
See simulator related issue, PR and demo:
- Issue: https://github.com/kubeedge/ianvs/issues/348
- Proposal Pull Request: https://github.com/kubeedge/ianvs/pull/35
- Code Pull Request: https://github.com/kubeedge/ianvs/pull/39
- Demo Video（in Chinese）on KEAW'22: https://www.bilibili.com/video/BV1nD4y1Y7cJ

**Pull request closed without merging**